src/classes/QueryBuilder.cls

```diff
@@ -70,107 +83,179 @@ public without sharing class QueryBuilder extends NebulaCore implements IQueryBu
-        String limitString = '';
-        if(this.limitCount != null) limitString = '\nLIMIT '+ this.limitCount;
-        return limitString;
+        return this.limitCount != null ? '\nLIMIT '+ this.limitCount : '';
     }
 
     private String getForUpdateString() {
-        String forUpdateString = '';
-        if(this.orderByList.isEmpty() && this.forUpdate) forUpdateString = '\nFOR UPDATE';
-        return forUpdateString;
+        return this.orderByList.isEmpty() && this.forUpdate ? '\nFOR UPDATE' : '';
     }
 
-    private String getUsingScopeString() {
-        String usingScopeString = '';
-        if(this.filterScope != null) usingScopeString = '\nUSING SCOPE ' + this.filterScope.name();
-        return usingScopeString;
-// TODO figure out what to do for SOSL https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_sosl_syntax.htm
+    private void setupAggregateQueryString(List<String> selectAliases, List<SObjectField> groupByFields,
+        List<QueryPredicate> wherePredicates, QueryAggregateCommand aggregationOperation) {
+        Integer groupByLength = groupByFields.size();
+        Integer aliasLength = selectAliases.size();
+        if(groupByLength != aliasLength) {
+            throw new Exceptions.InvalidArgumentException('Alias and group by fields must match');
+        }
+
+        this.setupStartOfSelect(groupByFields.get(0), selectAliases.get(0), aggregationOperation);
+        this.parseAdditionalSelectAndGroupByFields(groupByLength, selectAliases, groupByFields);
+        this.query += '\n' + this.getFromClauseString();
+        this.whereField(wherePredicates);
+        this.query += '\nGROUP BY ' + String.join(this.groupByList, ',');
+    }
+
+    private void setupStartOfSelect(SObjectField aggregationField, String aggregationAlias, QueryAggregateCommand aggregationOperation) {
+        this.query = 'SELECT ' + aggregationOperation + ' (' + aggregationField + ') ' + aggregationAlias + ', ';
     }
 
-    private String getQueryString() {
-        String query = 'SELECT ' + this.getQueryFieldString()
-            + '\nFROM ' + this.sobjectType
-            + this.getUsingScopeString()
+    private void parseAdditionalSelectAndGroupByFields(Integer groupByLength, List<String> selectAliases, List<SObjectField> groupByFields) {
+        List<String> queryStringSpecifiers = new List<String>();
+        //start at 1 because the first field will be the aggregationField
+        for(Integer i = 1; i < groupByLength; i++) {
+            String alias = selectAliases[i];
+            SObjectField field = groupByFields[i];
+            String fieldName = this.getFieldNameByRelationship(field);
+            this.groupByList.add(fieldName);
+            queryStringSpecifiers.add(fieldName + ' ' + alias);
+        }
+
+        this.query += String.join(queryStringSpecifiers, ',');
+    }
+
+    private String getFieldNameByRelationship(SObjectField field) {
```

Comment:
Mind removing this? I've got some other code that in progress that handles this a bit differently.